src/builders/CommandStep.ts

```diff
@@ -65,19 +72,70 @@ export class CommandStep implements StepBuilder, KeyBuilder, LabelBuilder, Condi
     this.#plugins.push(plugin)
     return this
   }
+  
+  parallelism(parallelism: number): this {
+    this.#parallelism = parallelism
+    return this
+  }
+  
+  env(env: Record<string, string | number>): this {
+    this.#env = env
+    return this
+  }
 
-  build(): CommandStepSchema {
-    if (!this.#command) {
-      throw new Error('CommandStep requires a command.')
+  addEnv(key: string, value: string | number): this {
+    if (!this.#env) {
+      this.#env = {}
+    }
+    this.#env[key] = value
+    return this
+  }
+
+  concurrency(jobs?: number, group?: string): this {
+    this.#concurrency = jobs
+    this.#concurrency_group = group
+    return this
+  }
+
+  softFail(fail: boolean = true): this {
+    this.#soft_fail = fail
+    return this
+  }
+
+  timeout(minutes: number): this {
+    this.#timeout_in_minutes = minutes
+    return this
+  }
+
+  artifactPath(path: string): this {
```

Comment:
nit/ I reckon we should leave this out and in the future add a plugin builder instead to canonicalise around https://github.com/buildkite-plugins/artifacts-buildkite-plugin

Reply:
✅